libraries/AP_Logger/AP_Logger.cpp

```diff
@@ -1241,6 +1247,59 @@ int16_t AP_Logger::Write_calc_msg_len(const char *fmt) const
     return len;
 }
 
+// thread for processing IO - in general IO involves a long blocking DMA write to an SPI device
+// and the thread will sleep while this completes preventing other tasks from running, it therefore
+// is necessary to run the IO in it's own thread
+void AP_Logger::io_thread(void)
+{
+    uint32_t last_sd_start_ms = AP_HAL::millis();
+    uint32_t last_run_us = AP_HAL::micros();
+
+    while (true) {
+        uint32_t now = AP_HAL::micros();
+
+        if (now - last_run_us < 1000) {
+            hal.scheduler->delay_microseconds(1000 - (now - last_run_us));
+        } else {
+            // always have some delay
+            hal.scheduler->delay_microseconds(250);
+        }
+
+        last_run_us = AP_HAL::micros();
+
+        FOR_EACH_BACKEND(io_timer());
+
+        now = AP_HAL::millis();
+
+        if (!hal.util->get_soft_armed()) {
+            // if sdcard hasn't mounted then retry it every 3s in the IO
+            // thread when disarmed
+            if (now - last_sd_start_ms > 3000) {
+                last_sd_start_ms = now;
+                AP::FS().retry_mount();
+            }
+        }
+    }
+}
+
+// start the update thread
+void AP_Logger::start_io_thread(void)
+{
+    WITH_SEMAPHORE(_log_send_sem);
+
+    if (_io_thread_started) {
+        return;
+    }
+
+    if (!hal.scheduler->thread_create(FUNCTOR_BIND_MEMBER(&AP_Logger::io_thread, void), "log_io", HAL_LOGGING_STACK_SIZE, AP_HAL::Scheduler::PRIORITY_IO, 0)) {
+        AP_HAL::panic("Failed to start Logger IO thread");
+        return;
```

Comment:
`panic` is `NORETURN` so you don't need `return`

Reply:
Fixed